lib/sinatra/base.rb

```diff
@@ -1057,13 +1057,16 @@ def route_missing
     # Attempt to serve static files from public directory. Throws :halt when
     # a matching file is found, returns nil otherwise.
     def static!(options = {})
-      return if (public_dir = settings.public_folder).nil?
-      path = File.expand_path("#{public_dir}#{URI_INSTANCE.unescape(request.path_info)}" )
-      return unless File.file?(path)
-
-      env['sinatra.static_file'] = path
+      public_dir = settings.public_folder
+      return if public_dir.nil?
+      path = "#{public_dir}#{URI_INSTANCE.unescape(request.path_info)}"
+      return unless valid_path?(path)
+      expanded_path = File.expand_path(path)
```

Comment:
@makushline can you stick with the `path` variable instead of introducing `exanded_path`?